src/swap.swaps/EthSwap.js

```diff
@@ -95,8 +95,10 @@ class EthSwap extends SwapInterface {
 
     debug('swap.core:swaps')('create after', this.gasPrice)
 
-    const base = BigNumber(10).pow(18)
-    const newAmount = new BigNumber(amount.toString()).times(base).integerValue().toNumber()
+
+    // const base = BigNumber(10).pow(18)
+    // const newAmount = BigNumber(amount).times(base).integerValue()
+    const newAmount = this.app.env.web3.utils.toWei(amount)
```

Comment:
С этой строкой будет так:

![screenshot_20190213_143801](https://user-images.githubusercontent.com/43396375/52687422-93598800-2f9d-11e9-8dd2-59f648b666d2.png)

Если сделать так, то будет всё ок:

```suggestion
    const newAmount = this.app.env.web3.utils.toWei(BigNumber(amount).toString())
```
![screenshot_20190213_144356](https://user-images.githubusercontent.com/43396375/52687489-e16e8b80-2f9d-11e9-847e-0e02b9a454c9.png)

Reply:
Странно, вроде amount был BN! Если он Number, то проблема еще хуже...

Reply:
x.toString() решает проблему

![screenshot_20190213_223640](https://user-images.githubusercontent.com/43396375/52712454-55ca1e80-2fe0-11e9-802f-cbbf9beda2da.png)